wordpress-mobile/WordPress-iOS
# Switch over most of the insights for new architecture.

Fixes #10888, #10889 (and more).

This is a counter-part to https://github.com/wordpress-mobile/WordPressKit-iOS/pull/88 — switching over fetching Insights to the new architecture (apart from the streak one, which was giving me some trouble — will follow-up in separate PR today or tomorrow).

Most of the code is straight-forward and changes are mostly due to different types we're using now to represent the data — and some shuffling of responsibilities around (no longer localizing strings in the networking layer, same for formatting dates, etc).

I also have 3/4 done persistence for this — I kept running into some weird issues so stripped the persistence from this and will PR it up separately this week.

To test: 
1. Open up Stats
2. Verify that the same Insights appear as before
3. Verify that the "Annual stats" insight card has data
4. Verify that the "Followers total" shows up correct sum of followers under the "Social" row where applicable
